concordium-node/src/grpc2.rs

```diff
@@ -2599,15 +2607,29 @@ where
                 // Time taken for the inner service to send back a response, meaning for
                 // streaming gRPC methods this is the duration for it to first return a stream.
                 let duration = request_received.elapsed().as_secs_f64();
-                if result.is_err() {
-                    grpc_request_duration
-                        .with_label_values(&[
-                            endpoint_name.as_str(),
-                            get_grpc_code_label(tonic::Code::Internal),
-                        ])
-                        .observe(duration);
+                match result {
+                    Err(e) => {
+                        // If the load shed service terminated the request this will be signalled as
+                        // an Err(Overloaded). So record resource exhaustion
+                        // in the metrics.
+                        let (code, response) = if e.is::<tower::load_shed::error::Overloaded>() {
+                            let new_response =
+                                tonic::Status::resource_exhausted("Too many concurrent requests.")
+                                    .to_http();
+                            (
+                                tonic::Code::ResourceExhausted,
```

Comment:
I think Unavailable may be a more appropriate status code. "The service is currently unavailable. This is most likely a transient condition, which can be corrected by retrying with a backoff."

Reply:
I based it on this table https://grpc.github.io/grpc/core/md_doc_statuscodes.html of example cases below.

```
Server temporarily out of resources (e.g., Flow-control resource limits reached) | RESOURCE_EXHAUSTED | Server
-- | -- | --
```